### PR TITLE
MVP and button functionality implemented

### PR DESCRIPTION
**Pull request by ezequielsetzesglobant**
The button was given functionality, where the button is pressed and a fixed number is set. 
The MVP was implemented. 
And the contracts were implemented.

Package structure
image

Image of the application running
image

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -36,4 +40,6 @@ dependencies {
     testImplementation 'junit:junit:4.+'
     androidTestImplementation 'androidx.test.ext:junit:1.1.2'
     androidTestImplementation 'androidx.test.espresso:espresso-core:3.3.0'
-}
\ No newline at end of file
+}
+
+
```
Comment (mauroLonghi) [inline app/build.gradle]: Pls remove empty spaces

Comment (SartiMau) [reply app/build.gradle]: leave only one here ;)

File: app/src/main/java/com/example/parkingsystem/mvp/model/ParkingModel.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+import com.example.parkingsystem.mvp.contract.ParkingContract;
+
+public class ParkingModel implements ParkingContract.Model {
+
+    private int number;
+
+    public void setNumber() {
+        this.number = 5;
+    }
+
```
Comment (SartiMau) [inline app/src/main/java/com/example/parkingsystem/mvp/model/ParkingModel.java]: remove this extra line pls

File: app/src/main/java/com/example/parkingsystem/mvp/presenter/ParkingPresenter.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
+package com.example.parkingsystem.mvp.presenter;
+
+import com.example.parkingsystem.mvp.contract.ParkingContract;
+import com.example.parkingsystem.mvp.model.ParkingModel;
+import com.example.parkingsystem.mvp.view.ParkingView;
+
+public class ParkingPresenter implements ParkingContract.Presenter {
+
+    private ParkingModel model;
```
Comment (SartiMau) [inline app/src/main/java/com/example/parkingsystem/mvp/presenter/ParkingPresenter.java]: you should use here the contract types

File: app/src/main/java/com/example/parkingsystem/mvp/presenter/ParkingPresenter.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    private ParkingView view;
+
+    public ParkingPresenter(ParkingModel model, ParkingView view) {
```
Comment (SartiMau) [inline app/src/main/java/com/example/parkingsystem/mvp/presenter/ParkingPresenter.java]: same here

File: app/src/main/java/com/example/parkingsystem/mvp/presenter/ParkingPresenter.java [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+        this.model = model;
+        this.view = view;
+    }
+
+    public void onSetParkingPlacesButtonPressed() {
+        model.setNumber();
+        view.showPopUp();
+    }
+
```
Comment (SartiMau) [inline app/src/main/java/com/example/parkingsystem/mvp/presenter/ParkingPresenter.java]: remove this extra line pls

File: app/src/main/java/com/example/parkingsystem/mvp/view/ParkingView.java [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+        this.binding = binding;
+    }
+
+    public void showPopUp() {
+        Context context = getContext();
+        if (context != null) {
+            Snackbar.make(binding.getRoot(), context.getResources().getString(R.string.snack_bar_button_click_message_main_activity), Snackbar.LENGTH_SHORT).show();
+        }
+    }
+
```
Comment (SartiMau) [inline app/src/main/java/com/example/parkingsystem/mvp/view/ParkingView.java]: remove this extra line pls

**Review by SartiMau — CHANGES_REQUESTED** [review]

**Review by lucianoeli — APPROVED** [review]

File: app/src/main/java/com/example/parkingsystem/mvp/view/ParkingView.java (hunk 6 above)
Comment (ezequielsetzesglobant) [reply app/src/main/java/com/example/parkingsystem/mvp/view/ParkingView.java]: Done

File: app/src/main/java/com/example/parkingsystem/mvp/presenter/ParkingPresenter.java (hunk 5 above)
Comment (ezequielsetzesglobant) [reply app/src/main/java/com/example/parkingsystem/mvp/presenter/ParkingPresenter.java]: Done

File: app/src/main/java/com/example/parkingsystem/mvp/presenter/ParkingPresenter.java (hunk 4 above)
Comment (ezequielsetzesglobant) [reply app/src/main/java/com/example/parkingsystem/mvp/presenter/ParkingPresenter.java]: Done

File: app/src/main/java/com/example/parkingsystem/mvp/presenter/ParkingPresenter.java (hunk 3 above)
Comment (ezequielsetzesglobant) [reply app/src/main/java/com/example/parkingsystem/mvp/presenter/ParkingPresenter.java]: Done

File: app/src/main/java/com/example/parkingsystem/mvp/model/ParkingModel.java (hunk 2 above)
Comment (ezequielsetzesglobant) [reply app/src/main/java/com/example/parkingsystem/mvp/model/ParkingModel.java]: Done

File: app/build.gradle (hunk 1 above)
Comment (ezequielsetzesglobant) [reply app/build.gradle]: Done

**Review by SartiMau — APPROVED** [review]

**Review by mauroLonghi — APPROVED** [review]

**Review by BQuinteros — APPROVED** [review]

**Pull request merged**